add translation for tooling page (#113)
vuejs-translations/docs-uk: src/guide/scaling-up/tooling.md

# Інструменти {#tooling}

## Спробуйте онлайн {#try-it-online}

> Вам не потрібно нічого встановлювати на свій комп’ютер, щоб випробувати одно-файлові компоненти Vue – існують онлайн-пісочниці, які дозволяють робити це прямо у браузері:

Reviewer: ```suggestion
Вам не потрібно нічого встановлювати на свій комп’ютер, щоб випробувати однофайлові компоненти Vue – існують онлайн пісочниці, які дозволяють робити це прямо у браузері:
```

- [Пісочниця Vue SFC](https://sfc.vuejs.org)
  - Завжди розгортається з останнього коміту

>   - Призначений для перевірки результатів компіляції компонентів

Reviewer: @winato прошу перечитувати гугл переклад. А краще — взагалі його використовувати по мінімуму, якщо можливо.
```suggestion
  - Призначена для перевірки результатів компіляції компонентів
```

- [Vue + Vite на StackBlitz](https://vite.new/vue)

>   - Середовище, схоже на IDE, у вебпереглядачі працює фактичний сервер Vite dev

Reviewer: ```suggestion
  - Середовище, схоже на IDE, що запускає фактичний сервер розробки Vite у браузері
```
Неправильна побудова речення. Гугл?

  - Найближчий до локального налаштування

> Також рекомендується використовувати ці онлайн-пісочниця, щоб надавати репродукції під час повідомлення про помилки.

Reviewer: ці онлайн-пісочниця
надавати репродукції

> Також рекомендується використовувати ці онлайн-пісочниці, для надання репродукцій при повідомленні про помилки.

Reviewer: ```suggestion
Також рекомендується використовувати ці онлайн пісочниці, для надання репродукцій при повідомленні про помилки.
```

## Побудова проекту {#project-scaffolding}

### Vite

> [Vite](https://vitejs.dev/) — це легкий і швидкий інструмент для створення з першокласною підтримкою Vue SFC. Його створив Еван Ю, який також є автором Vue!

Author: ```suggestion
[Vite](https://vitejs.dev/) — це легкий і швидкий інструмент для розробки з першокласною підтримкою Vue SFC. Його створив Еван Ю, який також є автором Vue!
```

Щоб розпочати роботу з Vite + Vue, просто запустіть:

<div class="language-sh"><pre><code><span class="line"><span style="color:var(--vt-c-green);">$</span> <span style="color:#A6ACCD;">npm init vue@latest</span></span></code></pre></div>

Ця команда встановить та виконає [create-vue](https://github.com/vuejs/create-vue), офіційний інструмент створення проектів Vue.

- Щоб дізнатися більше про Vite, перегляньте [документацію Vite](https://vitejs.dev).
- Щоб налаштувати спеціальну поведінку Vue у проекті Vite, наприклад, передати параметри компілятору Vue, перегляньте документи для [@vitejs/plugin-vue](https://github.com/vitejs/vite/tree/main/packages/plugin-vue#readme).

> Обидва онлайн-пісочниці, згадані вище, також підтримують завантаження файлів як проект Vite.

Reviewer: обидва пісочниці?

- Використовуйте [`vue-tsc`](https://github.com/johnsoncodehk/volar/tree/master/vue-language-tools/vue-tsc) для виконання перевірки типів з командного рядка або для створення файлів `d.ts` для SFC.

## Тестування {#testing}

Основна стаття: [посібник з тестування](/guide/scaling-up/testing).

- [Cypress](https://www.cypress.io/) рекомендовано для E2E тестів. Його також можна використовувати для тестування одно-файлових компонентів Vue за допомогою програми [Cypress Component Test Runner](https://Документація.cypress.io/guides/component-testing/introduction).

- [Vitest](https://vitest.dev/) – це програма для тестування, створена членами команди Vue / Vite, яка зосереджена на швидкості. Вона спеціально розроблена для додатків на основі Vite, щоб забезпечити миттєвий зворотний зв'язок для тестування частини/компонента.

- [Jest](https://jestjs.io/) можна змусити працювати з Vite через [vite-jest](https://github.com/sodatea/vite-jest). Однак це рекомендується, лише якщо у вас є існуючі набори тестів на основі Jest, які потрібно перенести на налаштування на основі Vite, оскільки Vitest надає подібні функції з набагато ефективнішою інтеграцією.

> ## Лінтинг {#linting}

Reviewer: ```suggestion
## Статичний аналіз {#linting}
```

> Команда Vue підтримує [eslint-plugin-vue](https://github.com/vuejs/eslint-plugin-vue), плагін [ESLint](https://eslint.org/), який підтримує спеціальне лінтування одно-файлових компонентів.

Reviewer: ```suggestion
Команда Vue підтримує [eslint-plugin-vue](https://github.com/vuejs/eslint-plugin-vue), плагін [ESLint](https://eslint.org/), який підтримує статичний аналіз спеціально для одно-файлових компонентів.
```

> Користувачі, які раніше використовували Vue CLI, можуть звикнути до конфігурації лінтера за допомогою Webpack. Однак, якщо використовується налаштування збірки на основі Vite, наша загальна рекомендація:

Reviewer: ```suggestion
Користувачі, які раніше використовували Vue CLI, можуть звикнути до конфігурації статичного аналізатора за допомогою Webpack. Однак, якщо використовується налаштування збірки на основі Vite, наша загальна рекомендація:
```

1. `npm install -D eslint eslint-plugin-vue`, потім дотримуйтеся [посібника з налаштування `eslint-plugin-vue`](https://eslint.vuejs.org/user-guide/#usage).

> 2. Налаштуйте розширення IDE ESLint, наприклад [ESLint для VSCode](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint), щоб ви отримували повідомлення від лінтера прямо у своєму редакторі під час розробки. Це також дозволяє уникнути зайвих витрат на перевірку під час запуску сервер розробки.

Reviewer: ```suggestion
2. Налаштуйте розширення IDE ESLint, наприклад [ESLint для VSCode](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint), щоб ви отримували повідомлення від статичного аналізатора прямо у своєму редакторі під час розробки. Це також дозволяє уникнути зайвих витрат на перевірку під час запуску сервер розробки.
```

> 3. Запустіть ESLint як частину команди виробничої збірки, щоб отримати повний відгук лінтера перед публікацією.

Reviewer: ```suggestion
3. Запустіть ESLint як частину команди виробничої збірки, щоб отримати повний звіт від статичного аналізатора перед публікацією.
```

> 4. (Необов’язково) Налаштуйте такі інструменти, як [lint-staged](https://github.com/okonet/lint-staged), щоб автоматично змінювати файли lint під час git-коміту.

Reviewer: ```suggestion
4. (Необов’язково) Налаштуйте такі інструменти, як [lint-staged](https://github.com/okonet/lint-staged), щоб автоматично змінювати статично проаналізовані файли під час git коміту.
```

## Форматування {#formatting}

> - Розширення VSCode [Volar](https://github.com/johnsoncodehk/volar) одразу забезпечує форматування для одно-файлових компонентів Vue.

Reviewer: ```suggestion
- Розширення VSCode [Volar](https://github.com/johnsoncodehk/volar) одразу забезпечує форматування для однофайлових компонентів Vue.
```

> - Крім того, [Prettier](https://prettier.io/) забезпечує вбудовану підтримку форматування одно-файлових компонентів Vue.

Reviewer: ```suggestion
- Крім того, [Prettier](https://prettier.io/) забезпечує вбудовану підтримку форматування однофайлових компонентів Vue.
```

## Інтеграції з користувацькими блоками SFC {#sfc-custom-block-integrations}

Користувацькі блоки компілюються в імпорт до того самого файлу Vue з різними запитами. Обробка цих імпортів залежить від основного інструменту збірки.

- Якщо використовується Vite, для перетворення відповідних користувальницьких блоків у виконуваний JavaScript слід використовувати спеціальний плагін Vite. [Приклад](https://github.com/vitejs/vite/tree/main/packages/plugin-vue#example-for-transforming-custom-blocks)

- Якщо використовується Vue CLI або звичайний Webpack, то Webpack має бути налаштований на перетворення відповідних блоків. [Приклад](https://vue-loader.vuejs.org/guide/custom-blocks.html)

## Пакети нижнього рівня {#lower-level-packages}

### `@vue/compiler-sfc`

- [Документація](https://github.com/vuejs/core/tree/main/packages/compiler-sfc)

> Цей пакет є частиною основного моно-репозиторію Vue і завжди публікується з тією ж версією, що й основний пакет `vue`. Він включений як залежний від основного пакета `vue` і проксі-сервер у `vue/compiler-sfc`, тому вам не потрібно встановлювати його окремо.

Reviewer: ```suggestion
Цей пакет є частиною основного моно репозиторію Vue і завжди публікується з тією ж версією, що й основний пакет `vue`. Він включений як залежний від основного пакета `vue` і проксі сервер у `vue/compiler-sfc`, тому вам не потрібно встановлювати його окремо.
```

> Сам пакет надає утиліти нижчого рівня для обробки одно-файлових компонентів Vue і призначений лише для авторів інструментів, яким потрібно підтримувати одно-файлові компоненти Vue у користувацьких інструментах.

Reviewer: ```suggestion
Сам пакет надає утиліти нижчого рівня для обробки однофайлових компонентів Vue і призначений лише для авторів інструментів, яким потрібно підтримувати однофайлові компоненти Vue у користувацьких інструментах.
```

:::tip
Завжди віддавайте перевагу використанню цього пакета через глибокий імпорт `vue/compiler-sfc`, оскільки це гарантує, що його версія синхронізується з середовищем виконання Vue.
:::

### `@vitejs/plugin-vue`

- [Документація](https://github.com/vitejs/vite/tree/main/packages/plugin-vue)

> Офіційний плагін, який забезпечує підтримку одно-файлових компонентів Vue у Vite.

Reviewer: ```suggestion
Офіційний плагін, який забезпечує підтримку однофайлових компонентів Vue у Vite.
```

### `vue-loader`

- [Документація](https://vue-loader.vuejs.org/)

> Офіційний завантажувач, який забезпечує підтримку одно-файлових компонентів Vue з Webpack. Якщо ви використовуєте Vue CLI, також перегляньте [документацію про зміну параметрів `vue-loader` у Vue CLI](https://cli.vuejs.org/guide/webpack.html#modifying-options-of-a-loader) .

Reviewer: ```suggestion
Офіційний завантажувач, який забезпечує підтримку однофайлових компонентів Vue з Webpack. Якщо ви використовуєте Vue CLI, також перегляньте [документацію про зміну параметрів `vue-loader` у Vue CLI](https://cli.vuejs.org/guide/webpack.html#modifying-options-of-a-loader) .
```

> ## Інші онлайн-пісочниці {#other-online-playgrounds}

Reviewer: ```suggestion
## Інші онлайн пісочниці {#other-online-playgrounds}
```
